File: src/types/translation.ts
export type TraditionalProviderKind =
  | 'google_translate'
  | 'deep_l'
  | 'azure_translator'
  | 'baidu'
  | 'youdao'
  | 'tencent_translator'
  | 'aliyun_translator'
  | 'custom';

export interface TraditionalTranslationProvider {
  type: 'traditional';
  id: string;
  name: string;
  kind: TraditionalProviderKind;
  endpoint?: string;
  extra: Record<string, string>;
}

export interface AiTranslationProvider {
  type: 'ai';
  id: string;
  name: string;
  aiProviderId: string;
  promptTemplate?: string;
}

export type TranslationProvider = TraditionalTranslationProvider | AiTranslationProvider;

export interface TranslationProviderSummary {
  id: string;
  name: string;
  providerType: string;
}

File: src-tauri/src/models/translation.rs
use std::collections::HashMap;

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(tag = "type", rename_all = "snake_case")]
pub enum TranslationProvider {
    Traditional {
        id: String,
        name: String,
        kind: TraditionalProviderKind,
        api_key_encrypted: Vec<u8>,
        endpoint: Option<String>,
        extra: HashMap<String, String>,
    },
    Ai {
        id: String,
        name: String,
        ai_provider_id: String,
        prompt_template: Option<String>,
    },
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum TraditionalProviderKind {
    GoogleTranslate,
    DeepL,
    AzureTranslator,
    Baidu,
    Youdao,
    TencentTranslator,
    AliyunTranslator,
    Custom,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TranslationProviderSummary {
    pub id: String,
    pub name: String,
    pub provider_type: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CachedTranslation {
    pub id: String,
    pub source_hash: String,
    pub target_lang: String,
    pub provider_id: String,
    pub translated_text: String,
    pub created_at: i64,
}
